source/dub/commandline.d

```diff
@@ -848,12 +849,55 @@ class BuildCommand : GenerateCommand {
 		args.getopt("f|force", &m_force, [
 			"Forces a recompilation even if the target is up to date"
 		]);
+		args.getopt("y|yes", &m_yes, [
+			`Automatic yes to prompts. Assume "yes" as answer to all prompts and run non-interactively.`
+		]);
 		super.prepare(args);
 		m_generator = "build";
 	}
 
 	override int execute(Dub dub, string[] free_args, string[] app_args)
 	{
+		// single package files don't need to be downloaded, they are on the disk.
+		if (free_args.length >= 1 && !m_single) {
+			auto package_name = free_args[0];
+			auto pack = dub.packageManager.getFirstPackage(package_name);
+
+			if (!pack) {
+				bool input(string caption, bool default_value = true) {
+					writef("%s [%s]: ", caption, default_value ? "Y/n" : "y/N");
+					auto inp = readln();
+					string userInput = "y";
+					if (inp.length > 1)
+						userInput = inp[0 .. $ - 1].toLower;
+
+					switch (userInput) {
+						case "no", "n", "0":
+							return false;
+						case "yes", "y", "1":
+						default:
+							return true;
+					}
+				}
+
+				// search for the package and filter for exact matches
+				auto search = dub.searchPackages(package_name)
+					.map!(tup => tup[1].find!(p => p.name == package_name))
+					.filter!(ps => !ps.empty);
+				if (search.empty)
+					return 2;
+
+				auto p = search.front.front;
+				writefln("%s wasn't found locally, but it's available online:", package_name);
+				writefln("Description: %s", p.description);
+				writefln("Version: %s", p.version_);
+
+				bool answer = m_yes ? true : input("Do you want to fetch %s?".format(package_name));
+				auto dep = Dependency(p.version_);
+				dub.fetch(package_name, dep, dub.defaultPlacementLocation, FetchOptions.none);
```

Comment:
Would be nice to be able to select the package repository and version, like with `docker run`, e.g.:
```
dub run ${PACKAGE_REGISTRY}/${PACKAGE_PATH}:${PACKAGE_VERSION}
```

Reply:
Nice idea :+1: 
This also removes the need for asking for verification in these cases. I only added support for selecting a specific version with `{PACKAGE_NAME}:{VERSION}` as selecting thee registry isn't really supported by `dub` at all at the moment.

Also note that e.g. `dub fetch` only supports `--version 1.0.4` (though it would be trivial to add support for `dub fetch mypackage:1.0.4`)

Reply:
So it turns out that the recently added `dub add` command uses `=`, but simply supporting both `=` and `:` will work.

> though it would be trivial to add support for dub fetch mypackage:1.0.4

https://github.com/dlang/dub/pull/1643